lib/synapse_pay_rest/models/subnet/subnet.rb

```
@@ -174,9 +174,121 @@ def lock
+        args = {
+          error: {
+            code:           response['error']['code'],
+            message:        response['error']['en'],
+            error_code:     response['error_code'],
+            http_code:      response['http_code']
+          }
+        }
+      else
+        args = {
+          transaction_id:   response['transaction_id'],
+          node_id:          response['node_id'],
+          subnet_id:        response['subnet_id']
+        }
+      end
+      
+      args
+    end
+
+    # Updates the given key value pairs.
+    # 
+    # @param card_pin [String]
+    # @param status [String]
+    # @param preferences [Hash]:
+    #   {
+    #     allow_foreign_transactions [Boolean],
+    #     daily_atm_withdrawal_limit [String, Integer],
+    #     daily_transaction_limit [String, Integer]
+    #   }
+    # 
+    # @raise [SynapsePayRest::Error] if HTTP error or invalid argument format
+    # 
+    # @return [SynapsePayRest::Subnet] new instance corresponding to same API record
+    def update(**options)
+      if options.empty?
+        raise ArgumentError, 'must provide a key-value pair to update. keys: card_pin,
+          status, preferences[:allow_foreign_transactions],
+          preferences[:daily_atm_withdrawal_limit], preferences[:daily_transaction_limit]'
+      end
+      response = node.user.client.subnets.update(
+        user_id: node.user.id,
+        node_id: node.id,
+        subnet_id: id,
+        payload: payload_for_card_update(options)
+      )
+      # return an updated subnet instance
+      self.class.from_response(node, response)
+    end
+
+
     # Checks if two Subnet instances have same id (different instances of same record).
     def ==(other)
       other.instance_of?(self.class) && !id.nil? && id == other.id
     end
+
+    private
+
+    # Converts #update args into API payload structure for CARD subnets.
+    def payload_for_card_update(**options)
+      payload = {}
+      # must have one of these
```

do we need to raise an error if at least one is not present?